lib/scheduler/variable_arrival_rate.go

```diff
@@ -0,0 +1,68 @@
+package scheduler
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/loadimpact/k6/lib/types"
+	null "gopkg.in/guregu/null.v3"
+)
+
+const variableArrivalRateType = "variable-arrival-rate"
+
+// VariableArrivalRateConfig stores config for the variable arrival-rate scheduler
+type VariableArrivalRateConfig struct {
+	BaseConfig
+	StartRate null.Int           `json:"startRate"`
+	TimeUnit  types.NullDuration `json:"timeUnit"` //TODO: rename to something else?
+	Stages    json.RawMessage    `json:"stages"`   //TODO: figure out some equivalent to stages?
```

Comment:
phases?

Reply:
I didn't mean the name :smile: though `phases` doesn't seem an improvement over `stages`...

Rather, I'm not sure if we should structure the contents of `stages` in the same way we currently have in the current k6 config and in the `VariableLoopingVUsConfig` - and array with a `{duration: X, target: Y}` values. Specifically, the `target` key name is my problem - it will be clearer if we call it `endRate` or something like that. But even then, it still doesn't seem the most intuitive way to specify those rates and the changes between the rates... So yeah, open to ideas.